docs/RELEASE.md

```diff
@@ -19,16 +19,29 @@ Publishing involves creating a release tag and creating a new Release on GitHub.
 
 ## Versioning
 
+1. Make sure that the `docs` include all necessary information included in the release (example [tag compare](https://github.com/kubernetes-sigs/secrets-store-csi-driver/compare/v0.0.21...master)).
+1. Create a new release branch `release-X.X`
```

Comment:
update `test-infra` for new tests on the new branch

Reply:
@aramase @tam7t 
Can we define a trigger in test-infra on `^release-` instead of `^release-X.X`. This way we can avoid updating test-infra for each release. The downside is it will kick build for each release branch. WDYT?

Reply:
Yeah! @aramase just made https://github.com/kubernetes/test-infra/pull/23229